backend/server/controllers/checklists.js

```diff
@@ -178,21 +178,30 @@ module.exports = {
           }
           return checklistItemCopy
         }))
-        const checklistItems = await Promise.all(checklistForCategoryIdFiltered.map((checklistItem, index) => ChecklistItem.upsert({
-          id: checklistItem.id,
-          name: checklistItem.name,
-          textWhenOn: checklistItem.textWhenOn,
-          textWhenOff: checklistItem.textWhenOff,
-          checkedPoints: checklistItem.checkedPoints,
-          uncheckedPoints: checklistItem.uncheckedPoints,
-          category,
-          checklistId: result[1].dataValues.id,
-          order: checklistOrder + index,
-          prerequisite: null, // map later; see below
-          minimumRequirement: checklistItem.minimumRequirement,
-          minimumRequirementMetIf: checklistItem.minimumRequirementMetIf,
-          minimumRequirementGradePenalty: checklistItem.minimumRequirementGradePenalty
-        }, { returning: true })))
+
+        const checklistUpsert = checklistForCategoryIdFiltered.map((checklistItem, index) => {
+          const obj = {
+            id: checklistItem.id,
+            name: checklistItem.name,
+            textWhenOn: checklistItem.textWhenOn,
+            textWhenOff: checklistItem.textWhenOff,
+            checkedPoints: checklistItem.checkedPoints,
+            uncheckedPoints: checklistItem.uncheckedPoints,
+            category,
+            checklistId: result[1].dataValues.id,
+            order: checklistOrder + index,
+            prerequisite: null, // map later; see below
+            minimumRequirement: checklistItem.minimumRequirement,
+            minimumRequirementMetIf: checklistItem.minimumRequirementMetIf,
+            minimumRequirementGradePenalty: checklistItem.minimumRequirementGradePenalty
+          }
+          if (obj.id === undefined) {
+            delete obj.id
+          }
+          return obj
+        })
+
+        const checklistItems = await Promise.all(checklistUpsert.map(async checklistItem => (checklistItem.id === undefined ? [await ChecklistItem.create(checklistItem, { returning: true })] : ChecklistItem.upsert(checklistItem, { returning: true }))))
```

Comment:
How is async / await used on this line? If checklist item has an id, await creation and then return array containing new checklist item, otherwise just return a promise for upsert?

Reply:
upsert should've had an await, but linter said that it was redundant. Whether it actually is is a separate question, although the application still seems to work fine